builtin/k8s/config_sourcer.go

```diff
@@ -126,6 +127,25 @@ func (cs *ConfigSourcer) read(
 		case *corev1.ConfigMap:
 			value, ok = d.Data[k8sReq.Key]
 
+		case *corev1.Secret:
+			var secretValue []byte
+
+			secretValue, ok = d.Data[k8sReq.Key]
+			sEnc := base64.StdEncoding.EncodeToString(secretValue)
+			decValue, err := base64.StdEncoding.DecodeString(sEnc)
```

Comment:
Why do you encode and then immediately decode? Probably need to comment whats going on here.